### PR TITLE
[WIP] New module utility - meraki

### PR DESCRIPTION
**Pull request by kbreit**
##### SUMMARY
This is a module in preparation for my Meraki modules. I have not submitted any of the modules that rely on this, but I’m creating the PR so it can be vetted early.

Many changes will be coming for this over the next few months as I develop modules and test further.

This utility will be responsible for all calls to the Meraki API. Additionally, it will handle idempotency checking.

The meraki_test module which is included in this will be deleted once a real module is merged into the devel branch.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
meraki

##### ANSIBLE VERSION
Devel

##### ADDITIONAL INFORMATION
None

### CODE REVIEW

File: lib/ansible/module_utils/network/meraki/meraki.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Ansible, but is an independent component
+
+# This particular file snippet, and this file snippet only, is BSD licensed.
```
Comment (dagwieers) [inline lib/ansible/module_utils/network/meraki/meraki.py:5]: I think there will be a problem with the license. IIRC modules in Ansible ought to be GPLv3+ in order to ship with Ansible. The other problem is that you add copyright to other people from the ACI team, and that code is GPLv3, so whatever copyright is held from there, cannot be turned into BSD.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/network/meraki/meraki.py (hunk 1 above)
Comment (kbreit) [reply lib/ansible/module_utils/network/meraki/meraki.py:5]: I looked at a bunch of module utilities and they all have the same stock license verbiage at the top. Each of them references BSD license and not GPLv3. 

I removed the others from the ACI team and left myself. That’ll come in a future push.

Comment (dagwieers) [reply lib/ansible/module_utils/network/meraki/meraki.py:5]: Forget about my statement, the module_utils is BSD ! I was confusing the modules with module_utils. Nevermind.

**Review by dagwieers — APPROVED** [review]

**Comment by kbreit** [comment]
I will be merging this PR with a different PR so I am closing this.

**Comment by kbreit** [comment]
I will be merging this PR with a different PR, so closing this one.

**Pull request closed without merging**